syntax = "proto3";

service TaskInfoTransport{
    // 函数没想好
}

// 信号和Signals.py中定义的不一样
enum Signal{
    SIGNAL_SEND = 0;      // 发送数据信号
    SIGNAL_RECV = 1;      // 接收数据信号
    SIGNAL_ACK = 2;       // 确认信号（未使用，可根据需要扩展）
    SIGNAL_CHECK = 3;     // 查询Cache信号
    SIGNAL_SKIP = 4;      // 跳过信号
    SIGNAL_TERMINATE = 9; // 终止信号
}

message IdTokenPair{
    int32 id = 1;
    string token_num = 2;
}

message TaskInfo {
    string request_id = 1;
    int32 id = 2;
    string infer_worker = 3;
    int32 token_num = 4;
    int32 data_length = 5;
    int32 index = 6;
    int32 task_type = 7;
    string type = 8;
    int32 task_num = 9;
    repeated int32 cache_pages_list = 10;
    repeated IdTokenPair id_token_pair = 11;
  }